includes/OAuthAuthenticator.php

```diff
@@ -0,0 +1,83 @@
+<?php
+/**
+ * OAuthAuthenticator
+ * Used to retrieve to authenticate the client & retrieves the access token
+ *
+ * @package GbWeiss
+ * @author Towa Digital <developer@towa.at>
+ * @license https://www.gnu.org/licenses/gpl-3.0.en.html GPL-3.0-or-later
+ */
+
+namespace GbWeiss\includes;
+
+use stdClass;
+
+/**
+ * OAuthAuthenticator Class
+ */
+class OAuthAuthenticator
+{
+    /**
+     * The endpoint used to authenticate against.
+     *
+     * @var string
+     */
+    private $authenticationEndpoint = null;
+
+    /**
+     * The http client implementation used.
+     *
+     * @var Object
+     */
+    private $client = null;
+
+    /**
+     * Constructor.
+     *
+     * @param Object $client Authentication Client implemenation.
+     */
+    public function __construct($client)
+    {
+        $this->client = $client;
+    }
+
+    /**
+     * Sets the authentication endpoint.
+     *
+     * @param string $endpoint The endpoint used to authenticate against.
+     * @return void
+     */
+    public function setAuthenticationEndpoint(string $endpoint)
+    {
+        $this->authenticationEndpoint = $endpoint;
+    }
+
+    /**
+     * Returns the access token
+     *
+     * @param string $clientId the id of the client.
+     * @param string $clientSecret the secret key of the client.
+     * @throws \Exception With message.
+     * @return string
+     */
+    public function authenticate(string $clientId, string $clientSecret): string
```

Comment:
@Martin-Welte I don't think that is neccessary, since this will not be passed to the plugin from the "outside". These settings can be fetched when needed.